src/cluster.c

```diff
@@ -1014,9 +1056,69 @@ getNodeByQuery(client *c, struct serverCommand *cmd, robj **argv, int argc, int
      * when writing a module that implements a completely different
      * distributed system. */
 
+    /* Determine transaction slot and return early on cross-slot. */
+    if (c->cmd->proc == execCommand && c->flag.multi) {
+        int slot = -1;
```

Comment:
maybe introduce a new `mc_slot` field in `client` and keep it updated after parsing every child command. this way we can avoid this loop, making the code easier to read and also less code to execute.